main.py

```diff
@@ -0,0 +1,45 @@
+from flask import Flask, render_template, request
+import random
+from dataclasses import dataclass
+
+
+app = Flask(__name__)
+
+
+@dataclass
+class User:
+    name: str
+    language: str
+    course: str = ''
+    grade: int = 0
+
+
+@app.route('/', methods=['GET', 'POST'])
+def home():
+    if request.method == 'POST':
+        name = request.form['name']
+        language = request.form['language']
+        user = User(name, language)
+        return render_template('course.html', user=user)
+    return render_template('index.html')
+
+
+@app.route('/course', methods=['POST'])
+def course():
+    user = User(request.form['name'], request.form['language'])
```

Comment:
that is not correct we should use user from the global context, as well as not asking user to add name and language on step when the course input only is expected